File: app/src/main/java/com/sample/demo/dao/ReportDao.java
package com.sample.demo.dao;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

import java.util.List;

public class ReportDao {

    @SerializedName("field-name")
    @Expose
    public String fieldName;
    @SerializedName("type")
    @Expose
    public String type;
    @SerializedName("min")
    @Expose
    public Integer min;
    @SerializedName("max")
    @Expose
    public Integer max;
    @SerializedName("required")
    @Expose
    public Boolean required;
    @SerializedName("options")
    @Expose
    public List<String> options = null;

}

File: app/src/main/java/com/shivam/demo/dao/ReportDao.kt
package com.shivam.demo.dao

import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName

import java.io.Serializable

class ReportDao : Serializable {

    @SerializedName("field-name")
    @Expose
    var fieldName: String? = null
    @SerializedName("type")
    @Expose
    var type: String? = null
    @SerializedName("min")
    @Expose
    var min: Int? = null
    @SerializedName("max")
    @Expose
    var max: Int? = null
    @SerializedName("required")
    @Expose
    var required: Boolean? = null
    @SerializedName("options")
    @Expose
    var options: List<String>? = null
    @SerializedName("fields")
    @Expose
    var fields: List<ReportDao>? = null

    var value:String? = null
}